common/djangoapps/student/tests/test_filters.py

```diff
@@ -235,3 +236,193 @@ def test_unenrollment_blocked_by_filter(self):
+            - The dashboard is rendered using the filtered enrollments list.
+        """
+        CourseEnrollment.enroll(self.user, self.first_course.id)
+        CourseEnrollment.enroll(self.user, self.second_course.id)
+
+        response = self.client.get(self.dashboard_url)
+
+        self.assertContains(response, self.first_course.id)
+        self.assertNotContains(response, self.second_course.id)
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.dashboard.render.started.v1": {
+                "pipeline": [
+                    "common.djangoapps.student.tests.test_filters.TestRenderAlternativeDashboard",
+                ],
+                "fail_silently": False,
+            },
+        },
+        PLATFORM_NAME="My site",
+    )
+    def test_dashboard_render_invalid(self):
+        """
+        Test rendering an invalid template after catching PreventDashboardRender exception.
+
+        Expected result:
+            - DashboardRenderStarted is triggered and executes TestRenderAlternativeDashboard.
+            - The server error template is rendered instead of the usual dashboard.
+        """
+        response = self.client.get(self.dashboard_url)
+
+        self.assertContains(response, "There has been a 500 error on the <em>My site</em> servers")
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.dashboard.render.started.v1": {
+                "pipeline": [
+                    "common.djangoapps.student.tests.test_filters.TestRedirectDashboardPageStep",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_dashboard_redirect(self):
+        """
+        Test redirecting to a new page after catching RedirectDashboardPage exception.
+
+        Expected result:
+            - DashboardRenderStarted is triggered and executes TestRedirectDashboardPageStep.
+            - The view response is a redirection.
+        """
+        response = self.client.get(self.dashboard_url)
+
+        self.assertEqual(status.HTTP_302_FOUND, response.status_code)
+
+    @override_settings(
+        OPEN_EDX_FILTERS_CONFIG={
+            "org.openedx.learning.dashboard.render.started.v1": {
+                "pipeline": [
+                    "common.djangoapps.student.tests.test_filters.TestRenderCustomResponse",
+                ],
+                "fail_silently": False,
+            },
+        },
+    )
+    def test_dashboard_custom_response(self):
+        """
+        Test returning a custom response after catching RenderCustomResponse exception.
+
+        Expected result:
+            - DashboardRenderStarted is triggered and executes TestRenderCustomResponse.
+            - The view response contains the custom response text.
+        """
+        response = self.client.get(self.dashboard_url)
+
+        self.assertContains(response, "This is a custom response.")
```

Comment:
Can go even harder here:

```suggestion
        self.assertEqual("This is a custom response.", response.content.decode("utf-8"))
```

Reply:
Thank you! :)